### ISSUE TITLE
Missing boot parameter/Adjust boot logic

### ISSUE THREAD

**Issue by scaronni** [openSUSE/nvidia-driver-G06#62]
Hi Stefan,

I just noticed we miss disabling the `nouveau` driver from the `initrd`. `nouveau` needs to be blacklisted in two places; when the root filesystem is mounted and modules get loaded from the filesystem, and when the system boots and the modules get loaded from the `initrd`.

If it's also not excluded from the `initrd`, by the time the root filesystem is mounted, the `nouveau` driver is already loaded.

Test systems in NVIDIA have `nouveau` already disabled by default, so I missed that, I spotted it while doing the ACL tests.

This, that we have already in the packages, takes care of blocking the modules when loading from the filesystem:
```
slaanesh@localhost:~> cat /etc/modprobe.d/50-nvidia.conf | grep ^blacklist
blacklist nouveau
```

For blocking loading of `nouveau` in the `initrd`, there are two options:

1. Boot loader parameter:
```
localhost:~ # cat /etc/default/grub | grep nouveau
GRUB_CMDLINE_LINUX_DEFAULT="splash=silent preempt=full mitigations=auto quiet security=apparmor rd.driver.blacklist=nouveau"
localhost:~ # grub2-mkconfig -o /boot/grub2/grub.cfg 
Generating grub configuration file ...
Found theme: /boot/grub2/themes/openSUSE/theme.txt
Found linux image: /boot/vmlinuz-6.4.0-150600.23.25-default
Found initrd image: /boot/initrd-6.4.0-150600.23.25-default
Warning: os-prober will be executed to detect other bootable partitions.
Its output will be used to detect bootable binaries on them and create new boot entries.
1514.890015 | DM multipath kernel driver not loaded
Adding boot menu entry for UEFI Firmware Settings ...
done
```

2. Exclusion from the `initrd`:
```
localhost:~ # cat /usr/lib/dracut/dracut.conf.d/60-nvidia.conf
omit_drivers+=" nvidia nvidia-drm nvidia-modeset nvidia-peermem nvidia-uvm nouveau "
```

Pros and cons of boot parameter:

- :green_circle: When booting in case of issues, one can remove the boot parameter and use `nouveau` while debugging.
- :green_circle: The `initrd` is never rebuilt, neither when installing the NVIDIA driver neither when uninstalling.
- :red_circle: The boot parameter needs to be added and removed when installing and uninstalling the NVIDIA driver.

Pros and cons of `initrd` module exclusions

- :green_circle: No need to manage the boot parameter when installing/uninstalling.
- :green_circle: No extra configuration files besides the configuration file for `modprobe` and for `dracut`.
- :red_circle: The `initrd` must be rebuilt every time the NVIDIA driver is installed or removed.

What's your preference?

**Comment by develop7** [comment]
The initrd: it is rebuilt quite often (on Tumbleweed, that is) already and managing a single file covered by package manager is way more reliable than fumbling with the kernel command line

**Comment by sndirsch** [comment]
I can do a PR later in the evening.

**Comment by scaronni** [comment]
> For some reason the nvidia kernel modules are added to initrd although we're already using dracut config snippet.
> 
> ```
> $ cat /etc/dracut.conf.d/60-nvidia.conf 
> omit_drivers+=" nvidia nvidia-drm nvidia-modeset nvidia-peermem nvidia-uvm "
> ```

This is definitely not normal... i'll check

**Comment by scaronni** [comment]
Actually, no, they are not added, I just checked.

We can also avoid the Grub parameter if we rebuild the initrd on install, so the `modprobe.d/50-nvidia.conf` is included in the initrd and . Which now brings me back to... if we need anyway to rebuild the initrd maybe we can just drop the `dracut` snippet and just let it rebuild the `initrd`?

**Comment by scaronni** [comment]
Let me make a merge request.

**Comment by scaronni** [comment]
Made a merge request, this should simplify even more. Please note that originally, the closed modules were always added to the `initrd`, while the open modules were added only on non-Optimus systems. That would have not been complete anyway as `nouveau` would get loaded anyway on Optimus systems.

I think this is much clearer and simpler. @sndirsch what do you think?

**Comment by sndirsch** [comment]
Now fixed with PR #66.
